### PR TITLE
ML-049/integration with the movie db

### PR DESCRIPTION
**Pull request by KarinaKlevzhits**
# Description

Add service for fill data base with TheMovieDB Api
Add rake task
Add job
Config TheMovieDB gem

## Trello card link

https://trello.com/c/bHi8AZNK/54-ml-049-themoviedb

# Checklist:

- [x] My code fixes the problem or adds required functionality
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove that my feature works
- [x] New and existing unit tests pass locally with my changes

# Additional requirements:

- [x] Does your code require any migrations to be ran?
- [x] Does your code require installing any new gems or system dependencies?
- [ ] Does your code require any changes to current deployment process?

# Additional Information

### CODE REVIEW

**Review by WhatWasThatt — APPROVED** [review]
Good job!

**Review by dmitrymalllpr — APPROVED** [review]

File: Gemfile.lock [hunk 1]
```diff
@@ -419,4 +434,4 @@ RUBY VERSION
    ruby 3.0.2p107
 
 BUNDLED WITH
-   2.3.5
+   2.2.22
```
Comment (ShrewHools) [inline Gemfile.lock:437]: please do not change the version unnecessarily

File: app/services/the_movie_db.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      company = { name: result['name'] }
+      Company.create(company)
+    end
+  end
+
+  def people_request
+    (1..LIMIT).each do |i|
+      result = Tmdb::Person.detail(i)
+      country = Country.find_or_create_by(name: result['place_of_birth']&.split&.last)
+      country&.save
```
Comment (ShrewHools) [inline app/services/the_movie_db.rb:40]: why?

File: app/services/the_movie_db.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      person = { name: result['name']&.split&.first, surname: result['name']&.split&.last,
+                 birthday: result['birthday'], deathday: result['deathday'], biography: result['biography'],
+                 country_id: country[:id],
```
Comment (ShrewHools) [inline app/services/the_movie_db.rb:43]: country.id?

File: app/services/the_movie_db.rb [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                 married: Faker::Boolean.boolean }
+      Person.create(person).save
```
Comment (ShrewHools) [inline app/services/the_movie_db.rb:45]: create and save?

ar = AR.new(name: params[:name])
ar.errors unless ar.save


read about using **save, new, create, save!, update** and else

File: app/services/the_movie_db.rb [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      movie = Movie.create(movie_values)
+      movie.languages.push(related_table_request(result['spoken_languages'], 'english_name', Language))
+      movie.genres.push(related_table_request(result['genres'], 'name', Genre))
+      movie.companies.push(related_table_request(result['production_companies'], 'name', Company))
+      movie.countries.push(related_table_request(result['production_countries'], 'name', Country))
+      movie.save
+    end
+  end
+
+  def actors_request
```
Comment (ShrewHools) [inline app/services/the_movie_db.rb:65]: this method looks very strange

File: app/services/the_movie_db.rb [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    person = Person.all
+    (1..Movie.count).each do |i|
+      next unless person[i]&.biography&.include?('actor' || 'actress')
+
+      actor = { actor_id: person[i].id, movie_id: Movie.find(i).id, role_name: Faker::Artist.name }
+      ActorRole.create(actor)
+    end
+  end
+
+  def movie_staffs_request(staff_type_name)
```
Comment (ShrewHools) [inline app/services/the_movie_db.rb:75]: this method looks very strange

File: app/services/the_movie_db.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      movie.companies.push(related_table_request(result['production_companies'], 'name', Company))
+      movie.countries.push(related_table_request(result['production_countries'], 'name', Country))
+      movie.save
+    end
+  end
+
+  def actors_request
+    person = Person.all
+    (1..Movie.count).each do |i|
+      next unless person[i]&.biography&.include?('actor' || 'actress')
```
Comment (ShrewHools) [inline app/services/the_movie_db.rb:68]: 'actor' || 'actress' always return 'actor'
%w[actor actress].include?(biography) maybe?

File: app/services/the_movie_db.rb [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  def calculate_age_limit(adult)
+    adult == true ? AGE_ADULT : Faker::Number.between(from: MIN_AGE, to: MAX_AGE)
+  end
+
+  def related_table_request(array, key, class_name)
+    temp = []
+    if array.nil?
+      temp = []
+    else
+      (0..array.length - 1).each do |i|
```
Comment (ShrewHools) [inline app/services/the_movie_db.rb:94]: look at the **upto** method

File: app/services/the_movie_db.rb [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    end
+  end
+
+  def calculate_age_limit(adult)
+    adult == true ? AGE_ADULT : Faker::Number.between(from: MIN_AGE, to: MAX_AGE)
+  end
+
+  def related_table_request(array, key, class_name)
+    temp = []
+    if array.nil?
```
Comment (ShrewHools) [inline app/services/the_movie_db.rb:91]: return [] unless array

?

**Review by WhatWasThatt — APPROVED** [review]
Really gj!

**Review by dmitrymalllpr — APPROVED** [review]

**Pull request merged**